pkg/vsphere/session/session.go

```diff
@@ -155,14 +155,10 @@ func (s *Session) Connect(ctx context.Context) (*Session, error) {
 		}
 	}
 
+	soapURL.User = s.User
 	// Update the service URL with expanded defaults
 	s.Service = soapURL.String()
```

Comment:
I am worried that removing this entirely will cause issues in other parts of the codebase w/o much time for deep testing.